src/engagement_db_to_analysis/engagement_db_to_analysis.py

```diff
@@ -72,12 +75,14 @@ def _get_project_messages_from_engagement_db(analysis_configurations, engagement
                 if latest_message_timestamp is None or msg_last_updated > latest_message_timestamp:
                     latest_message_timestamp = msg_last_updated
 
-            # Export latest message timestamp to cache
-            if latest_message_timestamp is not None or len(messages) > 0:
-                cache.set_latest_message_timestamp(engagement_db_dataset, latest_message_timestamp)
+            if cache is not None:
+                # Export latest message timestamp to cache
+                if latest_message_timestamp is not None:
+                    cache.set_latest_message_timestamp(engagement_db_dataset, latest_message_timestamp)
 
-            # Export project engagement_dataset files
-            cache.set_messages(engagement_db_dataset, messages)
+                # Export project engagement_dataset files
+                if len(messages) > 0:
```

Comment:
Why do we need this check?

Reply:
To check if there are messages to be set in the analysis cache; Does not allow the opening of the cache if there is no data.